packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -146,6 +146,12 @@ class ScrollPositionWithSingleContext extends ScrollPosition implements ScrollAc
     final Simulation simulation = physics.createBallisticSimulation(this, velocity);
     if (simulation != null) {
       beginActivity(BallisticScrollActivity(this, simulation, context.vsync));
+      if (simulation is ScrollSpringSimulation &&
```

Comment:
This is odd, why do we special-case ScrollSpringSimulation here?

"is" type-checks like this usually indicate a code smell and should be avoided.